File: scripts/add-weight-to-products.js
// Add weight to all products in menu-final.json
const fs = require('fs');

console.log('Adding weight to products...');

const data = JSON.parse(fs.readFileSync('menu-final.json', 'utf8'));

// Weight ranges by category (in grams)
const WEIGHT_RANGES = {
    'pizza-30cm': { min: 400, max: 600 },
    'piccolo-20cm': { min: 150, max: 250 },
    'calzone': { min: 300, max: 500 },
    'bread-focaccia-bread': { min: 200, max: 400 },
    'bread-focaccia-focaccia': { min: 250, max: 450 },
    'sauce': { min: 50, max: 150 },
    'rolls-sushi': { min: 30, max: 50 },
    'rolls-rolls': { min: 200, max: 350 },
    'combo': { min: 800, max: 1500 },
    'confectionery': { min: 100, max: 300 },
    'mors': { min: 250, max: 500 },
    'juice': { min: 200, max: 400 },
    'water': { min: 500, max: 500 },
    'soda': { min: 330, max: 500 },
    'beverages-other': { min: 200, max: 500 },
    'frozen': { min: 300, max: 600 },
    'aromatic-oils': { min: 100, max: 250 },
    'masterclass': { min: 0, max: 0 },
    'franchise': { min: 0, max: 0 }
};

let totalProducts = 0;
let productsWithWeight = 0;

// Process each category
Object.keys(data.menu).forEach(category => {
    const range = WEIGHT_RANGES[category] || { min: 200, max: 400 };
    
    data.menu[category].forEach(product => {
        totalProducts++;
        
        // Skip if weight already exists
        if (product.weight) {
            productsWithWeight++;
            return;
        }
        
        // Add random weight within range
        if (range.min === 0 && range.max === 0) {
            product.weight = 0;
        } else {
            product.weight = Math.floor(Math.random() * (range.max - range.min + 1)) + range.min;
        }
        productsWithWeight++;
    });
});

console.log(`Total products: ${totalProducts}`);
console.log(`Products with weight: ${productsWithWeight}`);

// Save updated menu
fs.writeFileSync('menu-final.json', JSON.stringify(data, null, 2));
console.log('✅ Weight added to all products!');

File: scripts/add-weight-to-products.py
#!/usr/bin/env python3
import json
import random

print('Adding weight to products...')

with open('menu-final.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Weight ranges by category (in grams)
WEIGHT_RANGES = {
    'pizza-30cm': (400, 600),
    'piccolo-20cm': (150, 250),
    'calzone': (300, 500),
    'bread-focaccia-bread': (200, 400),
    'bread-focaccia-focaccia': (250, 450),
    'sauce': (50, 150),
    'rolls-sushi': (30, 50),
    'rolls-rolls': (200, 350),
    'combo': (800, 1500),
    'confectionery': (100, 300),
    'mors': (250, 500),
    'juice': (200, 400),
    'water': (500, 500),
    'soda': (330, 500),
    'beverages-other': (200, 500),
    'frozen': (300, 600),
    'aromatic-oils': (100, 250),
    'masterclass': (0, 0),
    'franchise': (0, 0)
}

total_products = 0
products_with_weight = 0

# Process each category
for category, products in data['menu'].items():
    weight_range = WEIGHT_RANGES.get(category, (200, 400))
    
    for product in products:
        total_products += 1
        
        # Skip if weight already exists
        if 'weight' in product:
            products_with_weight += 1
            continue
        
        # Add random weight within range
        if weight_range == (0, 0):
            product['weight'] = 0
        else:
            product['weight'] = random.randint(weight_range[0], weight_range[1])
        products_with_weight += 1

print(f'Total products: {total_products}')
print(f'Products with weight: {products_with_weight}')

# Save updated menu
with open('menu-final.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)

print('✅ Weight added to all products!')
